miden-crypto/src/merkle/smt/partial.rs

```diff
@@ -287,6 +289,33 @@ impl Default for PartialSmt {
     }
 }
 
+// CONVERSIONS
+// ================================================================================================
+
+impl From<Smt> for PartialSmt {
+    fn from(smt: Smt) -> Self {
+        PartialSmt(smt)
+    }
+}
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for PartialSmt {
+    fn write_into<W: winter_utils::ByteWriter>(&self, target: &mut W) {
+        target.write(&self.0);
+    }
+}
+
+impl Deserializable for PartialSmt {
+    fn read_from<R: winter_utils::ByteReader>(
+        source: &mut R,
+    ) -> Result<Self, winter_utils::DeserializationError> {
+        let inner: Smt = source.read()?;
+        Ok(PartialSmt(inner))
+    }
+}
```

Comment:
I've merged this, but now that I think about it, there might be an issue: to store partial SMT data we use the full SMT, but when we serialize the full SMT, we only serialize the leaf nodes. In the case of partial SMT, I'm not sure this will be sufficient to properly deserialize it (i.e., we might have serialize both the leaves and the inner nodes).

@igamigo - could you create a test (or update an existing one) which does the round-trip serialization of a partial SMT which doesn't contain all leaf nodes?

Also, cc @PhilippGackstatter as the original author of the partial SMT.